### ISSUE TITLE
ValueError: LineStrings must have at least 2 coordinate tuples

### ISSUE THREAD

**Issue by Charlesfox1** [Toblerity/Shapely#524]
A code I am working returns "ValueError: LineStrings must have at least 2 coordinate tuples" 
when I pass this:

splitted = shapely.ops.split(line, points) 

...where line is a multilinestring object and points a multipoint object, as described in the attached zipfile (POINT_fails.csv, LINE_fails.csv). Any idea why this is happening? The same code 'works' on the version of the information tagged with underscore 'works'. .

If it helps, I am picking up and handling these objects from pandas with the shapely.wkt.loads functionality. I know I am passing two shapely objects to the function to being with. 

[example data.zip](https://github.com/Toblerity/Shapely/files/1315207/example.data.zip)

Full error trace:

```
Traceback (most recent call last):
  File "Network_clean.py", line 202, in <module>
    iteration_roads, iteration_junctions = RoadCleanup(Phase_1_roads, Phase_1_junctions, 1)
  File "Network_clean.py", line 194, in RoadCleanup
    splitted = shapely.ops.split(line, points)
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\ops.py", line 421, in split
    return GeometryCollection([i for part in geom.geoms for i in SplitOp.split(part, splitter).geoms])
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\ops.py", line 442, in split
    return GeometryCollection(split_func(geom, splitter))
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\ops.py", line 387, in _split_line_with_multipoint
    new_chunks.extend(SplitOp._split_line_with_point(chunk, pt))
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\ops.py", line 363, in _split_line_with_point
    LineString(coords[:i+1]),
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\geometry\linestring.py", line 49, in __init__
    self._set_coords(coordinates)
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\geometry\linestring.py", line 94, in _set_coords
    self._geom, self._ndim = geos_linestring_from_py(coordinates)
  File "C:\Users\charl\Anaconda2\lib\site-packages\shapely\geometry\linestring.py", line 258, in geos_linestring_from_py
    "LineStrings must have at least 2 coordinate tuples")
ValueError: LineStrings must have at least 2 coordinate tuples
```

**Comment by sgillies** [comment]
@Charlesfox1 interesting. I see why the case that succeeds does so: the split output is equivalent, as a whole and piece-wise, to the input. The other case crashes my interpreter after getting this warning from the GEOS lib: `IllegalArgumentException: point array must contain 0 or >1 elements`.

**Comment by Charlesfox1** [comment]
Thank you so much Sean. I can't see any syntactical differences between the two, so I assume it's some edge-case in the data i am supplying it causing it to behave this way... your attention is most appreciated!
Charles

**Comment by Charlesfox1** [comment]
Hey Sean - any movement on this? I only ask because the team goes to Viet Nam in 10 days to present the tool built on this, and I would love for it to work on beyond the 'confirmed case' data I provided. 

Best 
Charles

**Comment by sgillies** [comment]
I cannot promise a time frame for a fix.

**Comment by Charlesfox1** [comment]
That's fair enough. Do you happen to know of any other line-splitting functions / libs out there so we can implement a work around?

Best, 
Charles

**Comment by sgillies** [comment]
Closed via #533
